### ISSUE TITLE
Controller Prompts are a bit Jank

### ISSUE THREAD

**Issue by Algester** [KingKrouch/SvSFix#1]
so with force disable steam input in the SvSFix.cfg enabled the prompts on screen are stuck at "keyboard" mode, this verifies that the mod is somewhat "working"

so I disabled it and set the Controller Prompts Type = PS5
I still somehow get XBOX prompts on screen so I'm not sure if its supposed to be PS5 or DS4?

**Comment by Naitrate** [comment]
Yeah, I have the foundation for native PS4/PS5/Switch controller support up, but the problem with that is that I want to load the native prompts that the game uses for those which are still in the game files. Seemingly, loading from AssetBundles are a bit temperamental, so I gotta figure out a solution for this.

Sorry for the very tardy response.

**Comment by Coates087** [comment]
Hey, I just tested your updated SvSFix mod with my GOG version of the game and it works mostly fine, but there were two problems with it. First I had to delete the "steam_api64.dll" in the `<path-to-game>\neptunia-sisters-vs-sisters_Data\Plugins\x86_64\` directory in order to get the game running. Once I did that, I noticed that while everything else works, the button prompts are missing.
 
![image](https://github.com/KingKrouch/SvSFix/assets/157427682/997a141c-45c7-4618-8a46-bc5df82acbbd)

**Comment by Naitrate** [comment]
> Hey, I just tested your updated SvSFix mod with my GOG version of the game and it works mostly fine, but there were two problems with it. First I had to delete the "steam_api64.dll" in the `<path-to-game>\neptunia-sisters-vs-sisters_Data\Plugins\x86_64\` directory in order to get the game running. Once I did that, I noticed that while everything else works, the button prompts are missing.
> 
> ![image](https://private-user-images.githubusercontent.com/157427682/303193671-997a141c-45c7-4618-8a46-bc5df82acbbd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDczNTg5NTcsIm5iZiI6MTcwNzM1ODY1NywicGF0aCI6Ii8xNTc0Mjc2ODIvMzAzMTkzNjcxLTk5N2ExNDFjLTQ1YzctNDYxOC04YTQ2LWJjNWRmODJhY2JiZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMjA4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDIwOFQwMjE3MzdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yMTYwNTNlYTZiYmZmMTIwZDY2YTEyZWZkZjMxNmZkY2EwNjc2NWU0ZTE5MmZjYzQ2NmI4MmY1MTJlOTllMDk3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.OdFsb0Av4VkQF0dvbO1Mxc4TPJCXZ6eacBWnpvqvw8c)

Hi! I should be having a new update that will be coming soon that addresses a bunch of issues with the mod. One of these improvements is native support with DualShock 4 and DualSense prompts straight from the game files, alongside Japanese layout menu navigation. Switch prompts ultimately will depend on if I can get files from the extracted contents of the Switch version working on PC through AssetBundle files.

This issue probably might be due to SteamInput not being force disabled in the config files. I'll need to figure out a good approach of figuring out if the game is using a GOG version of the game, and prevent any of the Steamworks enhancements from running, but I imagine porting my mod to the IL2CPP version of the game will present with some issues as I manually updated the version of Steamworks.NET that the game uses, so I could even access the extra Steamworks functionality.

**Comment by Coates087** [comment]
> > Hey, I just tested your updated SvSFix mod with my GOG version of the game and it works mostly fine, but there were two problems with it. First I had to delete the "steam_api64.dll" in the `<path-to-game>\neptunia-sisters-vs-sisters_Data\Plugins\x86_64\` directory in order to get the game running. Once I did that, I noticed that while everything else works, the button prompts are missing.
> > ![image](https://private-user-images.githubusercontent.com/157427682/303193671-997a141c-45c7-4618-8a46-bc5df82acbbd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDczNTg5NTcsIm5iZiI6MTcwNzM1ODY1NywicGF0aCI6Ii8xNTc0Mjc2ODIvMzAzMTkzNjcxLTk5N2ExNDFjLTQ1YzctNDYxOC04YTQ2LWJjNWRmODJhY2JiZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMjA4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDIwOFQwMjE3MzdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yMTYwNTNlYTZiYmZmMTIwZDY2YTEyZWZkZjMxNmZkY2EwNjc2NWU0ZTE5MmZjYzQ2NmI4MmY1MTJlOTllMDk3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.OdFsb0Av4VkQF0dvbO1Mxc4TPJCXZ6eacBWnpvqvw8c)
> 
> Hi! I should be having a new update that will be coming soon that addresses a bunch of issues with the mod. One of these improvements is native support with DualShock 4 and DualSense prompts straight from the game files, alongside Japanese layout menu navigation. Switch prompts ultimately will depend on if I can get files from the extracted contents of the Switch version working on PC through AssetBundle files.
> 
> This issue probably might be due to SteamInput not being force disabled in the config files. I'll need to figure out a good approach of figuring out if the game is using a GOG version of the game, and prevent any of the Steamworks enhancements from running, but I imagine porting my mod to the IL2CPP version of the game will present with some issues as I manually updated the version of Steamworks.NET that the game uses, so I could even access the extra Steamworks functionality.

Thanks for the quick reply. Outside of the issues I've mentioned, your fix did seem to get rid of the camera issue for both versions of the game.

**Comment by Naitrate** [comment]
There's quite a few things that I want to do before I can say that I'm ready to have a new release of the mod ready to download:

* Fixing the SteamInput implementation, as it seemed to have broke itself within the span of the game's previous update.
* Fixing the Pillarbox/Letterbox actor that I plan to introduce to mask out fullscreen UI elements that weren't supposed to be visible with aspect ratios other than 16:9.
* Fixing some UI problems left from the game, and from some incorrect UI calculations for arbitrary aspect ratios.
* Fixing the Grass Density feature that I'm working on (for improved performance on the Steam Deck).
* Figuring out how to backport some of the Switch version content (Specifically Switch button prompts, and the changed title screen background) to the PC version.
* Figuring out a reliable hook for directional lights, so I can employ some fixes to the Z-rotation, so it doesn't cause problems with shadow cascades.
* Using reflection to get shadow resolutions higher than 4096 pixels or lower than 512 pixels working.
* Figuring out why forcing Anisotropic Filtering doesn't seem to do literally anything (It still looks like the game is using bilinear or trilinear filtering).

Once I can fix these things, and then add a few things that I want to add before I'm comfortable with a new release, I'll probably push an update out for the mod, alongside making another blog post detailing more technical information regarding the mod.

I felt kind of rushed to release this in time for the game's release, when I felt like I needed at least another two months of work to get things in a polished enough state where I'd be happy leaving it in maintenance mode. Finding things with the game was the time range around September to November of 2022, while actually making the plugin and fleshing out features was around two months of constant work and troubleshooting (Between December 2022 and February 2023). Been working on this for the past week or so, and the goal now is to actually finish this mod, and some of my other mods before I retire from making mods for games, and potentially use this kind of stuff on a resume.

If you want to give the current build a try, to test out the native PS controller support (Make sure that you already have SvSFix and the Mono Runtimes installed, alongside SteamInput disabled in the game properties on Steam before doing so), give this a try. It also includes the skippable intros and Japanese controller layout features:
[SvSFix_WIP_NativePSPrompts.zip](https://github.com/KingKrouch/SvSFix/files/14203863/SvSFix_WIP_NativePSPrompts.zip)

**Comment by KaitoSoft** [comment]
> There's quite a few things that I want to do before I can say that I'm ready to have a new release of the mod ready to download:
> 
>     * Fixing the SteamInput implementation, as it seemed to have broke itself within the span of the game's previous update.
> 
>     * Fixing the Pillarbox/Letterbox actor that I plan to introduce to mask out fullscreen UI elements that weren't supposed to be visible with aspect ratios other than 16:9.
> 
>     * Fixing some UI problems left from the game, and from some incorrect UI calculations for arbitrary aspect ratios.
> 
>     * Fixing the Grass Density feature that I'm working on (for improved performance on the Steam Deck).
> 
>     * Figuring out how to backport some of the Switch version content (Specifically Switch button prompts, and the changed title screen background) to the PC version.
> 
>     * Figuring out a reliable hook for directional lights, so I can employ some fixes to the Z-rotation, so it doesn't cause problems with shadow cascades.
> 
>     * Using reflection to get shadow resolutions higher than 4096 pixels or lower than 512 pixels working.
> 
>     * Figuring out why forcing Anisotropic Filtering doesn't seem to do literally anything (It still looks like the game is using bilinear or trilinear filtering).
> 
> 
> Once I can fix these things, and then add a few things that I want to add before I'm comfortable with a new release, I'll probably push an update out for the mod, alongside making another blog post detailing more technical information regarding the mod.
> 
> I felt kind of rushed to release this in time for the game's release, when I felt like I needed at least another two months of work to get things in a polished enough state where I'd be happy leaving it in maintenance mode. Finding things with the game was the time range around September to November of 2022, while actually making the plugin and fleshing out features was around two months of constant work and troubleshooting (Between December 2022 and February 2023). Been working on this for the past week or so, and the goal now is to actually finish this mod, and some of my other mods before I retire from making mods for games, and potentially use this kind of stuff on a resume.
> 
> If you want to give the current build a try, to test out the native PS controller support (Make sure that you already have SvSFix and the Mono Runtimes installed, alongside SteamInput disabled in the game properties on Steam before doing so), give this a try. It also includes the skippable intros and Japanese controller layout features: [SvSFix_WIP_NativePSPrompts.zip](https://github.com/KingKrouch/SvSFix/files/14203863/SvSFix_WIP_NativePSPrompts.zip)

I am using GOG as well and, while the buttons don't show up like the user above, the Circle/X swap does work. I want to report that a UI part is moved to left instead of right (I have already edited everything in the cfg but it stays on the left, while in works normally in the version from the release page):

![np](https://github.com/KingKrouch/SvSFix/assets/132292308/42560c3e-530b-4079-ae0a-e415360d711b)

By the way, I hope you will release a GOG compatible version soon. In alternative, it would be cool to have a version that does not change the buttons at all (but keeps the Circle/X swap) and you could just do it by replacing the resource files with that mod from PCGamesWiki.

**Comment by Naitrate** [comment]
> I am using GOG as well and, while the buttons don't show up like the user above, the Circle/X swap does work. I want to report that a UI part is moved to left instead of right (I have already edited everything in the cfg but it stays on the left, while in works normally in the version from the release page):

Yeah, the UI modifications that I'm doing are all sorts of fucked, and I think I'm gonna have to rewrite a ton of that. The developers decided to hardcode positions and use incorrect anchoring, which is causing problems with different display aspect ratios.

> By the way, I hope you will release a GOG compatible version soon. In alternative, it would be cool to have a version that does not change the buttons at all (but keeps the Circle/X swap) and you could just do it by replacing the resource files with that mod from PCGamesWiki.

I will probably have to eventually figure out how to port all of these changes (and accurately find a way to see what version of the game is being used) before it can be compatible with the GOG release.

**Comment by Coates087** [comment]
> > I am using GOG as well and, while the buttons don't show up like the user above, the Circle/X swap does work. I want to report that a UI part is moved to left instead of right (I have already edited everything in the cfg but it stays on the left, while in works normally in the version from the release page):
> 
> Yeah, the UI modifications that I'm doing are all sorts of fucked, and I think I'm gonna have to rewrite a ton of that. The developers decided to hardcode positions and use incorrect anchoring, which is causing problems with different display aspect ratios.
> 
> > By the way, I hope you will release a GOG compatible version soon. In alternative, it would be cool to have a version that does not change the buttons at all (but keeps the Circle/X swap) and you could just do it by replacing the resource files with that mod from PCGamesWiki.
> 
> I will probably have to eventually figure out how to port all of these changes (and accurately find a way to see what version of the game is being used) before it can be compatible with the GOG release.

I believe the GOG version of the game is the same version as the Steam version. Outside of compatibility for the DLC, I think the difference between the two are in the x86_64 folder. See below for details:
![image](https://github.com/KingKrouch/SvSFix/assets/157427682/dec52aed-a6fd-49be-ac0d-ef4c1efad677)

The top one is the GOG version, while the bottom one is the Steam version. Yes, I did buy both version of the game.

**Comment by Naitrate** [comment]
> I believe the GOG version of the game is the same version as the Steam version. Outside of compatibility for the DLC, I think the difference between the two are in the x86_64 folder. See below for details: ![image](https://private-user-images.githubusercontent.com/157427682/329068435-dec52aed-a6fd-49be-ac0d-ef4c1efad677.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTUyMTM4MDgsIm5iZiI6MTcxNTIxMzUwOCwicGF0aCI6Ii8xNTc0Mjc2ODIvMzI5MDY4NDM1LWRlYzUyYWVkLWE2ZmQtNDliZS1hYzBkLWVmNGMxZWZhZDY3Ny5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTA5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUwOVQwMDExNDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05YzNkZDU3NWMwNWRjMTVhMWVhMjRjNDAwNDNmMTFmZDZmMDM1OWVjYzU2MGU3NGUzNGMyMTg3MzZlNzAwMGRlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.3La6Umk_XJI9t3ZSGrhGsVbvUuL48JXzpwv0J44vzBE)
> 
> The top one is the GOG version, while the bottom one is the Steam version. Yes, I did buy both version of the game.

Thanks for this information, now I added the start of a check being ran for the GOG plugin in the specified directory, but now I gotta do some checks to disable the Steam enhancements and Steam functionality when the game starts.

![image](https://github.com/KingKrouch/SvSFix/assets/64479044/75addab8-2e9e-4c27-8b6f-dde7c5bfd95f)

Try giving this build a try (Which should disable the SteamInput and Steam enhancements for Photo Mode), keeping in mind to put the extracted files into BepInEx/Plugins:
[SvSFix_GOGPatches.zip](https://github.com/KingKrouch/SvSFix/files/15256157/SvSFix_GOGPatches.zip)

**Comment by Coates087** [comment]
> > I believe the GOG version of the game is the same version as the Steam version. Outside of compatibility for the DLC, I think the difference between the two are in the x86_64 folder. See below for details: ![image](https://private-user-images.githubusercontent.com/157427682/329068435-dec52aed-a6fd-49be-ac0d-ef4c1efad677.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTUyMTM4MDgsIm5iZiI6MTcxNTIxMzUwOCwicGF0aCI6Ii8xNTc0Mjc2ODIvMzI5MDY4NDM1LWRlYzUyYWVkLWE2ZmQtNDliZS1hYzBkLWVmNGMxZWZhZDY3Ny5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTA5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUwOVQwMDExNDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05YzNkZDU3NWMwNWRjMTVhMWVhMjRjNDAwNDNmMTFmZDZmMDM1OWVjYzU2MGU3NGUzNGMyMTg3MzZlNzAwMGRlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.3La6Umk_XJI9t3ZSGrhGsVbvUuL48JXzpwv0J44vzBE)
> > The top one is the GOG version, while the bottom one is the Steam version. Yes, I did buy both version of the game.
> 
> Thanks for this information, now I added the start of a check being ran for the GOG plugin in the specified directory, but now I gotta do some checks to disable the Steam enhancements and Steam functionality when the game starts.
> 
> ![image](https://private-user-images.githubusercontent.com/64479044/329086791-75addab8-2e9e-4c27-8b6f-dde7c5bfd95f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTUyMTQ5MzEsIm5iZiI6MTcxNTIxNDYzMSwicGF0aCI6Ii82NDQ3OTA0NC8zMjkwODY3OTEtNzVhZGRhYjgtMmU5ZS00YzI3LThiNmYtZGRlN2M1YmZkOTVmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTA5VDAwMzAzMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTdmNTRhYTIwMWFmZDA0NTg2MWJlZjhiNmRlM2Q4NDczZjNjMTg0NTI1ZTNmMmFjMGZhOWMyYmZiN2VhZGNhNTUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Elypctc4y4oVU0f8W2iAASGlbz7D8D5OJXQBgH6eDQA)
> 
> Try giving this build a try (Which should disable the SteamInput and Steam enhancements for Photo Mode), keeping in mind to put the extracted files into BepInEx/Plugins: [SvSFix_GOGPatches.zip](https://github.com/KingKrouch/SvSFix/files/15256157/SvSFix_GOGPatches.zip)

I really hate to say this, but I don't think it changed anything on my end. Sorry about that.

**Comment by Naitrate** [comment]
> I really hate to say this, but I don't think it changed anything on my end. Sorry about that.

Yeah, this is just the start. I'll have to get around to getting my hands on a GOG copy of the game to do further testing on, but I imagine that might have to wait until I port everything to support BepInEx IL2CPP.

**Comment by Coates087** [comment]
> > I really hate to say this, but I don't think it changed anything on my end. Sorry about that.
> 
> Yeah, this is just the start. I'll have to get around to getting my hands on a GOG copy of the game to do further testing on, but I imagine that might have to wait until I port everything to support BepInEx IL2CPP.

I would wait until the game gets on sale on GOG before getting it. I appreciate all of the hard work you are putting into your mod.

**Comment by KaitoSoft** [comment]
> > I really hate to say this, but I don't think it changed anything on my end. Sorry about that.
> 
> Yeah, this is just the start. I'll have to get around to getting my hands on a GOG copy of the game to do further testing on, but I imagine that might have to wait until I port everything to support BepInEx IL2CPP.

If you want I can send you my GOG version, GOG games are without DRM and can be launched without launcher.

By the way, you mentioned above that you got the file to swap X and Circle from the Switch release, right? Could you share that file please? Maybe it will work if I just drop it in the folder and replace the existing one. Honestly I would be happy with just a DS4 buttons mod (which I have already) and X and Circle swapped, since I know that porting the whole mod to GOG version requires time and effort.

**Comment by Naitrate** [comment]
> If you want I can send you my GOG version, GOG games are without DRM and can be launched without launcher.
> 
> By the way, you mentioned above that you got the file to swap X and Circle from the Switch release, right? Could you share that file please? Maybe it will work if I just drop it in the folder and replace the existing one. Honestly I would be happy with just a DS4 buttons mod (which I have already) and X and Circle swapped, since I know that porting the whole mod to GOG version requires time and effort.

I already have my hands on a GOG copy. Been busy with IT course work and getting my health back in shape.

As for the reverse X/A and Circle/B thing, that is a option in SvSFix.cfg when it generates an updated config file for the new build. It should automatically detect the controller type through Unity's input system or SteamInput and then apply the fix. For Switch controllers, it's forced on, as I wanted something consistent with that version of the game. Unsure if I have the build uploaded here.

For now, the mod only works with the Steam release. Once I port the fixes to work in IL2CPP, that's when I'm gonna do testing. Just that some things are difficult to achieve in IL2CPP BepInEx at the moment.

**Comment by Coates087** [comment]
I did discovered something a bit weird about this patch. When I tested the SvSFix on my GOG copy of the game and got to the cut scene "Remnants of the Organization" (near the beginning of Ch 2 at Leanbox), the game will go to a black screen, regardless of if the decensor patch is installed. The Steam version is able to play this scene with the SvSFix, both with and without the decensor patch. 

See attached for the saved files. The save file does contain spoilers since they are new game plus files.

[Chapter 2 save (new game plus).zip](https://github.com/user-attachments/files/15540615/Chapter.2.save.new.game.plus.zip)

**Comment by KaitoSoft** [comment]
> > If you want I can send you my GOG version, GOG games are without DRM and can be launched without launcher.
> > By the way, you mentioned above that you got the file to swap X and Circle from the Switch release, right? Could you share that file please? Maybe it will work if I just drop it in the folder and replace the existing one. Honestly I would be happy with just a DS4 buttons mod (which I have already) and X and Circle swapped, since I know that porting the whole mod to GOG version requires time and effort.
> 
> I already have my hands on a GOG copy. Been busy with IT course work and getting my health back in shape.
> 
> As for the reverse X/A and Circle/B thing, that is a option in SvSFix.cfg when it generates an updated config file for the new build. It should automatically detect the controller type through Unity's input system or SteamInput and then apply the fix. For Switch controllers, it's forced on, as I wanted something consistent with that version of the game. Unsure if I have the build uploaded here.
> 
> For now, the mod only works with the Steam release. Once I port the fixes to work in IL2CPP, that's when I'm gonna do testing. Just that some things are difficult to achieve in IL2CPP BepInEx at the moment.

The build you mentioned is uploaded above but I have only the game on GOG so no luck. I was asking if you had just the raw Switch file that tells the game what button does what. Or maybe I misunderstood and ther is no such file, and instead the swap is done by the plugin itself?

**Comment by Naitrate** [comment]
> Or maybe I misunderstood and ther is no such file, and instead the swap is done by the plugin itself?

It's being done with game code, so yeah, it's done by the plugin. There's a bit of input handling code that I'm modifying.

**Comment by AnubyteCode** [comment]
Some day...maybe this will work. It currently does nothing, doesn't even load, causes more problems than it solves. Though it does cause black screens. So it actually breaks things while fixing nothing. BepInEx doesn't even make a log file.

**Comment by Naitrate** [comment]
> Some day...maybe this will work. It currently does nothing, doesn't even load, causes more problems than it solves. Though it does cause black screens. So it actually breaks things while fixing nothing. BepInEx doesn't even make a log file.

What version of the game are you using (Steam, GOG, MS Store)? As of right now, it only supports Steam.

BIE not making a log file would indicate that it's not being loaded by the game, and that means that you are either using the wrong release of the game, and/or not using the Mono Runtimes.

I'm unsure what specific things have changed between the two other versions and the MS Store release, but since this mod technically relies on late beta stage Mono DLLs for the game, this was before those versions released.

**Comment by Naitrate** [comment]
> Some day...maybe this will work. It currently does nothing, doesn't even load, causes more problems than it solves. Though it does cause black screens. So it actually breaks things while fixing nothing. BepInEx doesn't even make a log file.

I saw your comment on the Steam forums, and as I can see next to the lack of a mouse cursor next to your name, you don't own the game on Steam. You're either using the MS Store, GOG version, or a pirated release, none of which are supported by this mod due to code differences with how those versions work (they are later releases than the one on Steam).

No need to harass me over this.

![Image](https://github.com/user-attachments/assets/1589fe38-6a71-4ec0-ad39-e86d723a6311)

**Comment by AnubyteCode** [comment]
Oh hey, you're around. I have the il2cpp version, yes. This fix is listed as essential on PCGamingWiki. I give most fault to the devs and Unity, this fix shouldn't actually be needed. 

It can have a bit of a chilling effect to tell people you're going to finish and then dripping updates without a release for so long. Someone else might have taken up the work, but are generally polite enough to let active coders finish their code.

Tried going into it myself, thinking I could compile the latest and do a port, but it's set up for your personalized dev environment. It's using .net standard 2 and some other ancient deps, at least one had a vulnerability. Managed to build the file pile but wasted hours of my day. I now resolve the stutter by temporarily setting my refresh rate to 50.

Good luck with this!
